services/grafana/auth_server.go

```diff
@@ -210,19 +235,29 @@ func (s *AuthServer) mustSetup(rw http.ResponseWriter, req *http.Request, l *log
 }
 
 // nextPrefix returns path's prefix, stopping on slashes and dots:
-// /foo.Bar/Baz -> /foo.Bar/ -> /foo. -> /
+// /inventory.Nodes/ListNodes -> /inventory.Nodes/ -> /inventory.Nodes -> /inventory. -> /inventory -> /
+// /v1/inventory/Nodes/List -> /v1/inventory/Nodes/ -> /v1/inventory/Nodes -> /v1/inventory/ -> /v1/inventory -> /v1/ -> /v1 -> /
 // That works for both gRPC and JSON URLs.
+// The chain ends with "/" no matter what.
 func nextPrefix(path string) string {
-	path = strings.TrimRight(path, "/.")
-	if i := strings.LastIndexAny(path, "/."); i != -1 {
-		return path[:i+1]
+	if len(path) == 0 || path[0] != '/' || path == "/" {
```

Comment:
Which case first symbol of `path` might be not equal to `/` in?
If it's possible why don't we handle paths like `foo.Bar/Baz`?

Reply:
It is not possible. Unless someone hacks nginx. :) Or we accidentally broke our nginx config.

Reply:
So, we did it only for `fuzz`? Why didn't we just add prefix `/` in `Fuzz`?